src/routes/products.js

```diff
@@ -32,11 +32,25 @@ router.post(
       .isEmpty()
       .withMessage('Please select the type of the product!'),
     body('location')
+      .isObject()
+      .withMessage(
+        'Location format is as follows; location{ lat: Number, lng: Number }'
+      )
       .not()
       .isEmpty()
-      .withMessage('Location cannot be empty!')
-      .isLength({ max: 50 })
-      .withMessage('Location cannot be more than 50 characters!'),
+      .withMessage('Location cannot be empty!'),
+    body('location.lat')
+      .not()
+      .isEmpty()
+      .withMessage('Latitude cannot be empty!')
+      .isNumeric()
+      .withMessage('Latitude should be a number!'),
+    body('location.lng')
+      .not()
+      .isEmpty()
+      .withMessage('Longitude cannot be empty!')
+      .isNumeric()
+      .withMessage('Longitude should be a number!'),
```

Comment:
Great work utilizing validator for specific object type 🔥

Reply:
Let's consider validating the values of `lat` and `lang` too.  Check this [guide](https://stackoverflow.com/questions/7780981/how-to-validate-latitude-and-longitude)

Reply:
Additional validation for `lat` and `lng` are added.

`(lat >= -90 && lat <= 90)`
`(lng >= -180 && lng <= 180)`